src/main/java/core/mate/academy/service/TruckProducer.java

```diff
@@ -0,0 +1,14 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Truck> {
```

Comment:
Ensure that `MachineProducer` is parameterized to prevent creation with non-Machine types as per checklist: Parameterize `MachineProducer` in such a way that any implementation NOT using `Machine` or its subtype could NOT be created.